wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/ast2wdlom/AstToTaskDefinitionElement.scala

```diff
@@ -1,12 +1,39 @@
 package wdl.draft3.transforms.ast2wdlom
 
-import common.Checked
-import common.validation.Checked._
+import cats.syntax.apply._
+import cats.syntax.validated._
+import cats.syntax.either._
+import common.collections.EnhancedCollections._
+import common.validation.ErrorOr._
 import wdl.draft3.parser.WdlParser.Ast
-import wdl.model.draft3.elements.TaskDefinitionElement
+import wdl.model.draft3.elements._
+import wdl.draft3.transforms.ast2wdlom.EnhancedDraft3Ast._
 
 object AstToTaskDefinitionElement {
 
-  def convert(a: Ast): Checked[TaskDefinitionElement] =
-    "FromAst[TaskDefinitionElement](a: Ast) is not implemented".invalidNelCheck
+  def convert(a: Ast): ErrorOr[TaskDefinitionElement] = {
+    val nameElementValidation: ErrorOr[String] = astNodeToString(a.getAttribute("name")).toValidated
+    val sectionsValidation: ErrorOr[Vector[TaskSectionElement]] = a.getAttributeAsVector[TaskSectionElement]("declarations").toValidated
+
+    (nameElementValidation, sectionsValidation) flatMapN combineElements
+  }
+
+  def combineElements(nameElement: String, bodyElements: Vector[TaskSectionElement]) = {
+    val inputsSectionElement: ErrorOr[Option[InputsSectionElement]] = validateSize(bodyElements.filterByType[InputsSectionElement], "inputs", 1)
+    val outputsSectionElement: ErrorOr[Option[OutputsSectionElement]] = validateSize(bodyElements.filterByType[OutputsSectionElement], "outputs", 1)
+    val runtimeSectionElement: ErrorOr[Option[RuntimeSectionElement]] = validateSize(bodyElements.filterByType[RuntimeSectionElement], "runtime", 1)
+
+    (inputsSectionElement, outputsSectionElement, runtimeSectionElement) mapN { (inputs, outputs, runtime) =>
+      TaskDefinitionElement(nameElement, inputs, outputs, runtime)
+    }
+  }
+
+  private def validateSize[A](elements: Vector[A], sectionName: String, numExpected: Int): ErrorOr[Option[A]] = {
```

Comment:
ToL: This name might be a little misleading, it should probably be something like `validatedSizeIsLessOrEqualTo` (since thinking ahead, the `command` is going to need an "exactly one" validation)